command/init_test.go

```diff
@@ -0,0 +1,206 @@
+package command
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/packer-plugin-sdk/acctest"
+	"golang.org/x/mod/sumdb/dirhash"
+)
+
+func TestInitCommand_Run(t *testing.T) {
+	// These tests will try to optimise for doing the least amount of github api
+	// requests whilst testing the max amount of things at once. Hopefully they
+	// don't require a GH token just yet. Acc tests are run on linux, darwin and
+	// windows, so requests are done 3 times.
+
+	type testCase struct {
+		checkSkip                             func(*testing.T)
+		name                                  string
+		inPluginFolder                        map[string]string
+		expectedPackerConfigDirHashBeforeInit string
+		hclFile                               string
+		packerConfigDir                       string
+		want                                  int
+		dirFiles                              []string
+		expectedPackerConfigDirHashAfterInit  string
+	}
+
+	cfg := &configDirSingleton{map[string]string{}}
+
+	tests := []testCase{
+		{
+			nil,
+			// here we pre-write plugins with valid checksums, Packer will
+			// see those as valid installations it did.
+			// the directory hash before and after init should be the same,
+			// that's a no-op. This also should do no GH query, so it is best
+			// to always run it.
+			"already-installed-no-op",
+			map[string]string{
+				"github.com/sylviamoss/comment/packer-plugin-comment_v0.2.18_x5.0_darwin_amd64":                "1",
+				"github.com/sylviamoss/comment/packer-plugin-comment_v0.2.18_x5.0_darwin_amd64_SHA256SUM":      "6b86b273ff34fce19d6b804eff5a3f5747ada4eaa22f1d49c01e52ddb7875b4b",
+				"github.com/sylviamoss/comment/packer-plugin-comment_v0.2.18_x5.0_windows_amd64.exe":           "1.exe",
+				"github.com/sylviamoss/comment/packer-plugin-comment_v0.2.18_x5.0_windows_amd64.exe_SHA256SUM": "07d8453027192ee0c4120242e6e84e2ca2328b8e0f506e2f818a1a5b82790a0b",
+				"github.com/sylviamoss/comment/packer-plugin-comment_v0.2.18_x5.0_linux_amd64":                 "1.out",
+				"github.com/sylviamoss/comment/packer-plugin-comment_v0.2.18_x5.0_linux_amd64_SHA256SUM":       "59031c50e0dfeedfde2b4e9445754804dce3f29e4efa737eead0ca9b4f5b85a5",
+			},
+			"h1:Q5qyAOdD43hL3CquQdVfaHpOYGf0UsZ/+wVA9Ry6cbA=",
```

Comment:
This is like a directory hash for the whole plugin directory; just to make sure nothing else is added in there.